packages/create-redwood-app/src/create-redwood-app.js

```diff
@@ -315,8 +246,114 @@ import { name, version } from '../package'
+        task: () => {
+          return new Promise((resolve) => {
+            const { engines } = require(path.join(templateDir, 'package.json'))
+
+            // this checks all engine requirements, including Node.js and Yarn
+            checkNodeVersion(engines, (_error, result) => {
+              if (result.isSatisfied) {
+                hasPassedEngineCheck = true
+                return resolve()
+              }
+              const logStatements = Object.keys(result.versions)
+                .filter((name) => !result.versions[name].isSatisfied)
+                .map((name) => {
+                  const { version, wanted } = result.versions[name]
+                  return style.error(
+                    `${name} ${wanted} required, but you have ${version}`
+                  )
+                })
+              engineErrorLog = logStatements
+              hasPassedEngineCheck = false
+              return resolve()
+            })
+          })
+        },
+      },
+    ],
+    { rendererOptions: { clearOutput: true } }
+  ).run()
+
+  // Show a success message if required engines are present
+  if (hasPassedEngineCheck === true) {
+    console.log(style.success(`✔️ Compatability checks passed.`))
+  }
+
+  // Show an error and prompt if failed engines check
+  if (hasPassedEngineCheck === false) {
+    console.log(style.error(`✖️ Compatability checks failed.`))
+    console.log(`${engineErrorLog.join('\n')}`)
+    console.log(
+      style.error(
+        `\nThis might make your RedwoodJS project incompatible with some deploy targets.`
+      )
+    )
+    console.log(style.header(`\nRelated documentation:`))
+    console.log(style.warning(`${engineErrorDocsLink}\n`))
+    const response = await prompt({
+      type: 'select',
+      name: 'override-engine-error',
+      message: 'How would you like to proceed?',
+      choices: [
+        { title: 'Override error and continue install', value: true },
+        { title: 'Quit install', value: false },
+      ],
+      initial: 0,
+    })
+    if (response['override-engine-error'] === false) {
+      process.exit(1)
+    }
+  }
+
+  // Main install Listr
   new Listr(
     [
+      {
+        title: 'Configuring installation preferences',
+        task: async (ctx, task) =>
+          task.newListr([
+            {
+              title: 'Language preference',
+              skip: () => typescript !== null,
+              task: async (ctx, task) => {
+                ctx.language = await task.prompt({
+                  type: 'Select',
+                  choices: ['TypeScript', 'JavaScript'],
+                  message: 'Select your preferred coding language',
+                  initial: 'TypeScript',
+                })
+                if (ctx.language === 'TypeScript') {
+                  task.output = 'TypeScript selected'
+                }
+                if (ctx.language === 'JavaScript') {
+                  task.output = 'JavaScript selected'
+                }
```

Comment:
Those are the only two options, so we can just write whichever's chosen:
```suggestion

                task.output = ctx.language
```